packages/react-html/README.md

```diff
@@ -12,112 +12,267 @@ $ yarn add @shopify/react-html
 
-## Interface
+`createSerializer()` accepts a single string argument for the identifier to use; this will help you find the serialized `script` tag if you need to debug later on. It also accepts a generic type argument for the type of the data that will be serialized/ available after deserialization.
 
-```typescript
-export interface Props {
-  children?: React.ReactNode;
-  styles?: Asset[];
-  scripts?: Asset[];
-  blockingScripts?: Asset[];
-  headData?: {[id: string]: any};
-  data?: {[id: string]: any};
-  hideForInitialLoad?: boolean;
-}
+The function returns a pair of components:
 
-interface Asset {
-  path: string;
-  integrity?: string;
-}
+```tsx
+createSerializer<string>('MyData');
 
-interface Browser {
-  userAgent: string;
-  supported: boolean;
-}
+// Would create the following components:
+
+function Serialize({data}: {data(): string}): null;
+function WithSerialized({
+  children,
+}: {
+  children(data: string | undefined): React.ReactNode;
+}): React.ReactNode;
 ```
 
-### Basic props
+The general pattern for using these components is to render the `WithSerialized` component as the top-most child. Within the render prop, construct whatever stateful store/ manager you need, using the data that was retrieved in cases where the serialization was found (on the browser, or on subsequent server renders). Finally, render the UI that depends on that stateful part, and a `Serialize` component that extracts the part that you you need to communicate between server and client.
 
-Most simple applications will only need these basic properties.
+Here is a complete example, using `@shopify/react-i18n`’s support for async translations as the data that needs to be serialized:
 
-**children**
-The children to be rendered inside the `#app` div.
+```tsx
+import {createSerializer} from '@shopify/react-html';
+import {Provider, Manager} from '@shopify/react-i18n';
 
-**styles**
-Descriptors for any style tags you want to include in the HEAD of the document.
+interface Props {
+  locale: string;
+  children: React.ReactNode;
+}
 
-**scripts**
-Descriptors for any script tags you want to include in your document. All scripts passed to this property will be deferred by appending them to the end of the document. We encourage this as a default, although you may use `blockingScripts` for any scripts that must be included in the HEAD of the document.
+const {Serialize, WithSerialized} = createSerializer<
+  ReturnType<Manager['extract']>
+>('i18n');
+
+export default function I18n({locale, children}: Props) {
+  return (
+    <WithSerialized>
+      {translations => {
+        const manager = new Manager({locale}, translations);
+
+        return (
+          <>
+            <Provider manager={manager}>{children}</Provider>
+            <Serialize
```

Comment:
Should we render something that uses a loaded piece of data in here?

ATM it still feels a little contrived as we load some data but never use it.

Reply:
The data is used to construct the manager? We'd be digging into a completely separate library's usage to show how i18n is used, Kaelig's comment was that this was already too confusing being tied to a different library :/